terraform/anomaly_detection/main.tf

```diff
@@ -0,0 +1,213 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+locals {
+  dataflow_service_account = "my-dataflow-sa"
+  max_dataflow_workers     = 1
+  worker_disk_size_gb      = 200
+  machine_type             = "g2-standard-4"
+}
+
+
+// Project
+module "google_cloud_project" {
+  source          = "github.com/GoogleCloudPlatform/cloud-foundation-fabric//modules/project?ref=v32.0.0"
+  billing_account = var.billing_account
+  project_create  = var.project_create
+  name            = var.project_id
+  parent          = var.organization
+  services = [
+    "cloudbuild.googleapis.com",
+    "dataflow.googleapis.com",
+    "monitoring.googleapis.com",
+    "pubsub.googleapis.com",
+    "autoscaling.googleapis.com",
+    "artifactregistry.googleapis.com",
+    "bigtable.googleapis.com",
+    "bigtableadmin.googleapis.com",
+    "bigquery.googleapis.com",
+  ]
+}
+
+module "registry_docker" {
+  source     = "github.com/GoogleCloudPlatform/cloud-foundation-fabric//modules/artifact-registry?ref=v32.0.0"
+  project_id = module.google_cloud_project.project_id
+  location   = var.region
+  name       = "dataflow-containers"
+  iam = {
+    "roles/artifactregistry.admin" = [
+      "serviceAccount:${module.google_cloud_project.number}@cloudbuild.gserviceaccount.com"
+    ]
+    "roles/artifactregistry.reader" = [
+      module.dataflow_sa.iam_email
+    ]
+  }
+  cleanup_policy_dry_run = false
+  cleanup_policies = {
+    keep-3-versions = {
+      action = "KEEP"
+      most_recent_versions = {
+        keep_count = 3
+      }
+    }
+  }
+}
+
+// Buckets for staging data, scripts, etc, in the two regions
+module "buckets" {
+  source        = "github.com/GoogleCloudPlatform/cloud-foundation-fabric//modules/gcs?ref=v32.0.0"
+  project_id    = module.google_cloud_project.project_id
+  name          = module.google_cloud_project.project_id
+  location      = var.region
+  storage_class = "STANDARD"
+  force_destroy = var.destroy_all_resources
+}
+
+module "input_topic" {
+  source     = "github.com/GoogleCloudPlatform/cloud-foundation-fabric//modules/pubsub?ref=v32.0.0"
+  project_id = module.google_cloud_project.project_id
+  name       = "messages"
```

Comment:
If this is going to be the input to the pipeline, I would adapt the name (to something like "transactions").